python/tests/test_linear_model.py

```diff
@@ -426,14 +429,19 @@ def test_linear_regression_spark_compat(
 
 
 @pytest.mark.parametrize("params_exception", params_exception)
-def test_fail_run_on_1_col(params_exception: Tuple[Dict[str, Any], bool]) -> None:
+def test_fail_run_on_1_col(
+    gpu_number: int, params_exception: Tuple[Dict[str, Any], bool]
+) -> None:
+    # reduce the number of GPUs for toy dataset to avoid empty partition
+    gpu_number = min(gpu_number, 1)
```

Comment:
All tests should work for the 2 gpu case, to have some coverage for distributed case.